File: bindings/python/examples/how_tos/alias_wallet/request_funds.py
from iota_sdk import Wallet, Utils, SyncOptions, AliasSyncOptions
from dotenv import load_dotenv
import os
import time

# In this example we request funds to an alias wallet.

load_dotenv()

FAUCET_URL = os.environ.get(
    'FAUCET_URL', 'https://faucet.testnet.shimmer.network/api/enqueue')

wallet = Wallet(os.environ['WALLET_DB_PATH'])

account = wallet.get_account('Alice')
balance = account.sync(None)

total_base_token_balance = balance.baseCoin.total
print(
    f'Balance before requesting funds on alias address: {total_base_token_balance}')

alias_id = balance.aliases[0]
print(f'Alias Id: {alias_id}')

# Get Alias address
alias_address = Utils.alias_id_to_bech32(
    alias_id, wallet.get_client().get_bech32_hrp())
faucet_response = wallet.get_client().request_funds_from_faucet(
    FAUCET_URL, alias_address)
print(faucet_response)

time.sleep(10)

sync_options = SyncOptions(alias=AliasSyncOptions(basic_outputs=True))

total_base_token_balance = account.sync(sync_options).baseCoin.total
print(
    f'Balance after requesting funds on alias address: {total_base_token_balance}')

File: bindings/nodejs/examples/how_tos/alias_wallet/request-funds.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Utils, Wallet, initLogger } from '@iota/sdk';

// This example uses secrets in environment variables for simplicity which should not be done in production.
//
// Make sure that `example.stronghold` and `example.walletdb` already exist by
// running the `how_tos/accounts_and_addresses/create-account` example!
//
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./how_tos/alias-wallet/request-funds.ts

// In this example we request funds to an alias wallet.
async function run() {
    initLogger();
    for (const envVar of ['WALLET_DB_PATH', 'FAUCET_URL']) {
        if (!(envVar in process.env)) {
            throw new Error(`.env ${envVar} is undefined, see .env.example`);
        }
    }
    try {
        const faucetUrl = process.env.FAUCET_URL as string;

        // Create the wallet
        const wallet = new Wallet({
            storagePath: process.env.WALLET_DB_PATH,
        });

        // Get the account we generated with `create_wallet`
        const account = await wallet.getAccount('Alice');

        const balance = await account.sync();

        const totalBaseTokenBalance = balance.baseCoin.total;
        console.log(
            `Balance before requesting funds on alias address: ${totalBaseTokenBalance}`,
        );

        const aliasId = balance.aliases[0];
        console.log(`Alias Id: ${aliasId}`);

        // Get Alias address
        const aliasAddress = Utils.aliasIdToBech32(
            aliasId,
            await (await wallet.getClient()).getBech32Hrp(),
        );
        const faucetResponse = await (
            await wallet.getClient()
        ).requestFundsFromFaucet(faucetUrl, aliasAddress);
        console.log(faucetResponse);

        await new Promise((resolve) => setTimeout(resolve, 10000));

        const syncOptions = {
            alias: {
                basicOutputs: true,
            },
        };
        const totalBaseTokenBalanceAfter = (await account.sync(syncOptions))
            .baseCoin.total;
        console.log(
            `Balance after requesting funds on alias address: ${totalBaseTokenBalanceAfter}`,
        );
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
